<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yuka Takemon" />


<title>Analyses</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">



<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <!-- NOTE: add "navbar-inverse" class for an alternate navbar background -->
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">The Aging Kidney</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="about.html">About</a></li>
        <li><a href="data.html">Data</a></li>
        <li><a href="analysis.html">Analyses</a></li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li><a href="https://github.com/TheJacksonLaboratory/TheAgingKidney">GitHub</a></li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analyses</h1>
<h4 class="author"><em>Yuka Takemon</em></h4>
<h4 class="date"><em>05 October, 2018</em></h4>

</div>


<p><strong>TODO LIST:</strong> * upload data and scripts * change paths to uploaded data</p>
<p><br> <br> <br></p>
<div id="overall-study-design" class="section level1">
<h1><span class="header-section-number">1</span> Overall study design</h1>
<p>We collected kidneys from 188 Diversity Outbred, 93 females and 95 males, mice at 6, 12, and 18 months of age (63, 62, and 63 animals respectively). The right kidney was homogenized and an aliquot was used for RNA-sequencing and another for shotgun proteomics. A week prior to sacrifice, we also collected urine samples for renal physiological phenotyping (albumin, phosphate, and creatinine levels).</p>
<p><img src="images/StudyDesign.png" width="80%" /></p>
</div>
<div id="mrna-and-protein-that-change-with-age" class="section level1">
<h1><span class="header-section-number">2</span> mRNA and protein that change with age</h1>
<p>We fitted a linear model for all 6,667 genes with both mRNA and protein expression information.</p>
<p><span class="math display">\[ y_{i} \sim Age + Sex + Generation \]</span> From linear model, we recorded slope and p-value generated from each mRNA and protein in the downstream processes. Additionally for a subset of genes with both mRNA and protein information, we also extracted the slope coefficient from.</p>
<div id="running-linear-regression-script" class="section level2">
<h2><span class="header-section-number">2.1</span> Running linear regression script</h2>
<p>The following bash script is designed to submit an Rscript to a cluster that utilizes PBS submission. You must submit the .Rdata and name the output file in this order.</p>
<p>The files needed are provided <span style="color:red"> HERE:</span></p>
<ul>
<li>DO188b_kidney_noprobs.RData</li>
<li>anova_tests_slope_pairs.R</li>
<li>Rsubmit_args.sh</li>
</ul>
<pre class="bash"><code># Make sure correct paths and used and paste below into cluster to run:

qsub -v I=&quot;./Rdata/DO188b_kidney_noprobs.RData ./Anova_output/kidney_anova_slope_output.csv&quot;,script=anova_tests_slope_pairs Rsubmit_args.sh</code></pre>
</div>
<div id="histograms-of-p-values" class="section level2">
<h2><span class="header-section-number">2.2</span> Histograms of p-values</h2>
<p>The figures below visualizes the distribution of p-values that incdicates whether the gene have significant changes in mRNA/protein as a result of age, sex, and age:sex interaction.</p>
<p>We found that sex effect on mRNA was stronger than the effects of Age, which is to be expected. We also saw that the proteins seem to be strongly affected by both sex and age.</p>
<div id="effects-on-mrna" class="section level3">
<h3><span class="header-section-number">2.2.1</span> Effects on mRNA</h3>
<p><img src="analysis_files/figure-html/ANOVA_sex_pval-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="effects-on-protein" class="section level2">
<h2><span class="header-section-number">2.3</span> Effects on protein</h2>
<p><img src="analysis_files/figure-html/ANOVA_age_pval-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="effects-of-agesex-interaction" class="section level2">
<h2><span class="header-section-number">2.4</span> Effects of Age:Sex interaction</h2>
<p><img src="analysis_files/figure-html/ANOVA_agesexInt_pval-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br> <br> <br></p>
</div>
</div>
<div id="direction-of-change-with-age" class="section level1">
<h1><span class="header-section-number">3</span> Direction of Change with Age</h1>
<p>We can illustrate the direction of change in mRNA and protein expression with age using the slope calculated from the linear model in the previous section.</p>
<p><img src="analysis_files/figure-html/SlopePlot-1.png" width="480" style="display: block; margin: auto;" /></p>
<p><br> <br></p>
<p>In our manuscript, we used <a href="http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a> to identify enriched biological functions and pathways of each of the 4 significant cluster of genes. For more information please refer to our manuscript <span style="color:red"> HERE. (needs link)</span></p>
</div>
<div id="the-ecological-fallacy" class="section level1">
<h1><span class="header-section-number">4</span> The Ecological Fallacy</h1>
<p>Previously we showed how mRNA and protein expression changed with age. We want to highlight a few genes to explain what is known as “The ecological fallacy”, whereby the correlation of the total population is not the same as individual age group correlation.</p>
<p><img src="images/EcoFallacy_v2.png" width="90%" /></p>
<p>We picked <em>Slc5a12</em> and <em>Flot1</em> as the best examples to demonstrate this ecological fallacy.</p>
<p>At the whole population level, the mRNA expression of <em>Slc5a12</em> decreased with age and the protein expression increased with age. However, when broken down by age groups, the mRNA and protein expressions are positively correlated. When were compare between the age groups we do not see a shift between 6 month and 12 months, but at 18 months of age the average of the group shifts to the top right. The upward shift indicates that the protein expression increased but the leftward shift indicates the decrease of mRNA expression.</p>
<p>In contrast to <em>Slc5a12</em>, <em>Flot1</em> is an example where by the mRNA expression increases with age and the protein expression decreases with age at the whole population level. At the level of the individual age groups, we observe a bottom-right shift with an increase in age. The downward shift indicates a decrease in protein expression, and a rightward shift indicates a increase in mRNA expression with age. <br></p>
<p><img src="analysis_files/figure-html/EcoFallacy_examples-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br> <br> <br></p>
</div>
<div id="relating-change-in-mrna-and-protein-to-renal-physiology" class="section level1">
<h1><span class="header-section-number">5</span> Relating change in mRNA and protein to renal physiology</h1>
<p>In addition to gene expression and protein expression quantification, we also measured metabolites in the urine samples (albumin, phosphate, and creatinine) collected from the same 188 animals 1 week prior to their sacrifice. From the analysis conducted in previous sections, we identified <em>Slc34a1</em>, <em>Slc34a3</em>, and <em>Lrp2</em> in group D to be of particular interest. Genes in group D have decreasing mRNA expression and increasing protein expression increasing with age at the whole population level.</p>
<p><br></p>
<div id="slc34a1-and-slc34a3-expression-levels" class="section level2">
<h2><span class="header-section-number">5.1</span> <em>Slc34a1</em> and <em>Slc34a3</em> expression levels</h2>
<div id="global-level" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Global level</h3>
<p><em>Slc34a1</em> and <em>Slc34a3</em> each encode a phosphate co-transporter (NPT2C and NPT2A) in localized in the proximal tubule for phosphate reabsorption. For both genes we observe we confirmed global expression levels of decreasing mRNA and increasing protein levels for both females and males. <br> <img src="analysis_files/figure-html/Slc34%20expression%20with%20age%20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="individual-age-group-level" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Individual age group level</h3>
<p><br> We also observe positive correlation between mRNA and protein levels at individual age groups. Both <em>Slc34a1</em> and <em>Slc34a3</em> show a shift towards the top-left corner with increasing age, which is refletive of the observations made at the whole population level. <br></p>
<p><img src="analysis_files/figure-html/Slc34%20mrna%20vs%20protein%20expression-1.png" width="672" style="display: block; margin: auto;" /> <br> As expected for mediators of phosphate reabsorption, <em>Slc34a1</em> and <em>Slc34a3</em> each encode a phosphate co-transporter (NPT2C and NPT2A) in localized in the proximal tubule. For both genes we observe we confirmed global expression levels of decreasing mRNA and increasing protein levels for both females and males. <br></p>
</div>
</div>
<div id="slc34a1-and-slc34a3-compared-to-phosphate" class="section level2">
<h2><span class="header-section-number">5.2</span> <em>Slc34a1</em> and <em>Slc34a3</em> compared to phosphate</h2>
<p><img src="analysis_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /> <br> <br></p>
</div>
<div id="lrp2-and-albumin" class="section level2">
<h2><span class="header-section-number">5.3</span> Lrp2 and Albumin</h2>
<p><img src="analysis_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br> <br> <br></p>
</div>
</div>
<div id="session-information" class="section level1">
<h1><span class="header-section-number">6</span> Session Information</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] bindrcpp_0.2.2  ggsci_2.8       gridExtra_2.3   forcats_0.3.0  
##  [5] stringr_1.3.0   dplyr_0.7.5     purrr_0.2.4     readr_1.1.1    
##  [9] tidyr_0.8.0     tibble_1.4.2    ggplot2_3.0.0   tidyverse_1.2.1
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_0.2.4 reshape2_1.4.3   haven_1.1.1      lattice_0.20-35 
##  [5] colorspace_1.3-2 htmltools_0.3.6  yaml_2.1.19      rlang_0.2.1     
##  [9] pillar_1.2.2     foreign_0.8-70   glue_1.3.0       withr_2.1.2     
## [13] modelr_0.1.1     readxl_1.1.0     bindr_0.1.1      plyr_1.8.4      
## [17] munsell_0.5.0    gtable_0.2.0     cellranger_1.1.0 rvest_0.3.2     
## [21] psych_1.8.3.3    evaluate_0.10.1  labeling_0.3     knitr_1.20      
## [25] parallel_3.5.1   broom_0.4.4      Rcpp_0.12.18     scales_1.0.0    
## [29] backports_1.1.2  jsonlite_1.5     mnormt_1.5-5     hms_0.4.2       
## [33] digest_0.6.15    stringi_1.2.4    grid_3.5.1       rprojroot_1.3-2 
## [37] cli_1.0.0        tools_3.5.1      magrittr_1.5     lazyeval_0.2.1  
## [41] crayon_1.3.4     pkgconfig_2.0.1  xml2_1.2.0       lubridate_1.7.4 
## [45] assertthat_0.2.0 rmarkdown_1.9    httr_1.3.1       rstudioapi_0.7  
## [49] R6_2.2.2         nlme_3.1-137     compiler_3.5.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
